relayer/go.mod

```diff
@@ -7,34 +7,49 @@ require (
 	github.com/btcsuite/btcd/chaincfg/chainhash v1.0.1
 	github.com/dontpanicdao/caigo v0.3.1-0.20220812122711-b855f2b57bb5
 	github.com/pkg/errors v0.9.1
+	github.com/prometheus/client_golang v1.12.2
 	github.com/smartcontractkit/chainlink-relay v0.1.6-0.20220824173548-dab4ac0a6595
 	github.com/smartcontractkit/chainlink-starknet/ops v0.0.0-20220822173222-d6c1c758059c
 	github.com/smartcontractkit/libocr v0.0.0-20220812191430-db92a9fdaa52
 	github.com/stretchr/testify v1.8.0
+	go.uber.org/multierr v1.8.0
 	golang.org/x/exp v0.0.0-20220608143224-64259d1afd70
 	gopkg.in/guregu/null.v4 v4.0.0
 )
 
 require (
 	github.com/avast/retry-go v3.0.0+incompatible // indirect
 	github.com/benbjohnson/clock v1.1.0 // indirect
+	github.com/beorn7/perks v1.0.1 // indirect
 	github.com/btcsuite/btcd/btcec/v2 v2.2.0 // indirect
+	github.com/cespare/xxhash/v2 v2.1.2 // indirect
+	github.com/confluentinc/confluent-kafka-go v1.8.2 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/deckarep/golang-set v1.8.0 // indirect
 	github.com/decred/dcrd/dcrec/secp256k1/v4 v4.0.1 // indirect
 	github.com/ethereum/go-ethereum v1.10.21 // indirect
 	github.com/go-ole/go-ole v1.2.6 // indirect
 	github.com/go-stack/stack v1.8.1 // indirect
+	github.com/golang/protobuf v1.5.2 // indirect
+	github.com/golang/snappy v0.0.4 // indirect
 	github.com/google/go-querystring v1.1.0 // indirect
 	github.com/google/uuid v1.3.0 // indirect
 	github.com/gorilla/websocket v1.5.0 // indirect
 	github.com/kr/text v0.2.0 // indirect
+	github.com/linkedin/goavro v2.1.0+incompatible // indirect
+	github.com/linkedin/goavro/v2 v2.9.7 // indirect
 	github.com/mattn/go-colorable v0.1.12 // indirect
 	github.com/mattn/go-isatty v0.0.14 // indirect
+	github.com/matttproud/golang_protobuf_extensions v1.0.1 // indirect
 	github.com/mr-tron/base58 v1.2.0 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
+	github.com/prometheus/client_model v0.2.0 // indirect
+	github.com/prometheus/common v0.35.0 // indirect
+	github.com/prometheus/procfs v0.7.3 // indirect
+	github.com/riferrei/srclient v0.4.1-0.20211229125508-8edc580da179 // indirect
```

Comment:
just some more thinking about how to structure monitoring components: maybe it's better to have a separate monitoring module outside of `relayer`?

pros:
* reduce the number of dependencies within the relayer module

cons:
* some dependency weirdness similar to how `ops` => `relayer` => `e2e_tests`, so then it would be `relayer` is a module used in `monitoring`

Reply:
This can be done. It would be easy to do here, but it would have to be done in chainlink-solana for consistency and other there it's harder the main go module is in the root of the project.

Reply:
makes sense! my thinking is that we use starknet implementation as a way to pilot the best layout of files/components that we would want to see in the other existing integrations like solana. we have some future planning to refactor the solana (and terra?) repos so we would propagate the best practices from starknet to the other chains

Reply:
I agree. I moved everything in `/monitoring`. See https://github.com/smartcontractkit/chainlink-starknet/pull/103/commits/7c822bc7787c845f9270648f525843ab457bd9f6